astropy/nddata/nduncertainty.py

```diff
@@ -378,3 +396,6 @@ def propagate_divide(self, other_nddata, result_data):
              result_data)
 
         return result_uncertainty
+
+uncertainty_type_dict = {item.uncertainty_type:item
+                         for item in NDUncertainty.__subclasses__()}
```

Comment:
Is there a reason for this to be a public, module-level variable?  It seems to me also that if you want a registry of uncertainty types a registry system would be more consistent with other similar code in Astropy.  See for example the time format registry: https://github.com/astropy/astropy/blob/master/astropy/time/core.py#L1298